### ISSUE TITLE
Crash in CompletionQueue.next()

### ISSUE THREAD

**Issue by rochaudhari** [grpc/grpc#30032]
What version of gRPC and what language are you using?
1.46.3

What operating system (Linux, Windows,...) and version?
Linux

What runtime / compiler are you using (e.g. python version or version of gcc)
Gcc

What did you do?
If possible, provide a recipe for reproducing the error. Try being specific and include code snippets if helpful.

We use the libuv based iomgr.

Waiting on completion queue(next) crashes randomly. Trace:

`grpc_combiner_continue_exec_ctx () at external/com_github_grpc_grpc/src/core/lib/iomgr/combiner.cc:188
188     external/com_github_grpc_grpc/src/core/lib/iomgr/combiner.cc: No such file or directory.
(gdb) bt
#0  grpc_combiner_continue_exec_ctx() () at external/com_github_grpc_grpc/src/core/lib/iomgr/combiner.cc:188
#1  0x000055567a71593a in grpc_core::ExecCtx::Flush() (this=0x7f771d7fd010) at external/com_github_grpc_grpc/src/core/lib/iomgr/exec_ctx.cc:82
#2  0x000055567a6e7a71 in end_worker(grpc_pollset*, grpc_pollset_worker*, grpc_pollset_worker**)
    (pollset=pollset@entry=0x55567deb08b8, worker=worker@entry=0x7f771d7fcee0, worker_hdl=worker_hdl@entry=0x0)
    at external/com_github_grpc_grpc/src/core/lib/iomgr/exec_ctx.h:233
#3  0x000055567a6e8e33 in pollset_work(grpc_pollset*, grpc_pollset_worker**, grpc_core::Timestamp) (ps=<optimized out>, worker_hdl=<optimized out>, deadline=...)
    at external/com_github_grpc_grpc/src/core/lib/iomgr/ev_epoll1_linux.cc:1059
#4  0x000055567a6e31fc in pollset_work(grpc_pollset*, grpc_pollset_worker**, grpc_core::Timestamp) (pollset=0x55567deb08b8, worker=0x0, deadline=...)
    at external/com_github_grpc_grpc/src/core/lib/iomgr/ev_posix.cc:321
#5  0x000055567a6968f8 in cq_next(grpc_completion_queue*, gpr_timespec, void*) (cq=0x55567deb07b0, deadline=..., reserved=<optimized out>)
    at external/com_github_grpc_grpc/src/core/lib/surface/completion_queue.cc:1047
#6  0x000055567a20c7a5 in grpc::CompletionQueue::AsyncNextInternal(void**, bool*, gpr_timespec)
    (this=this@entry=0x7f764fc3bad0, tag=tag@entry=0x7f771d7fd110, ok=ok@entry=0x7f771d7fd118, deadline=...)
    at external/com_github_grpc_grpc/src/cpp/common/completion_queue_cc.cc:146
#7  0x000055567a1a48e0 in grpc::CompletionQueue::Next(void**, bool*) (ok=0x7f771d7fd118, tag=0x7f771d7fd110, this=0x7f764fc3bad0)
`

What did you expect to see?
No crash

What did you see instead?
Crash as mentioned above.

Make sure you include information that can help us debug (full error message, exception listing, stack trace, logs).

See [TROUBLESHOOTING.md](https://github.com/grpc/grpc/blob/master/TROUBLESHOOTING.md) for how to diagnose problems better.

**Comment by rochaudhari** [comment]
Completion queue has write tag waiting and this happens while receving write tag

**Comment by cosmin42** [comment]
> This was a side effect of another issue in the code.

Could you link that issue here? I encounter the same problem on v1.48

**Comment by rochaudhari** [comment]
this has been before few weeks, I am think it was due to same tag multiple times, there was a race condition in my code which was leading to this. Do you mind sharing your code here, I can give it a shot.

**Comment by rochaudhari** [comment]
> > This was a side effect of another issue in the code.
> 
> Could you link that issue here? I encounter the same problem on v1.48

this has been before few weeks, I am think it was due to same tag multiple times, there was a race condition in my code which was leading to this. Do you mind sharing your code here, I can give it a shot.

**Comment by cosmin42** [comment]
It seems to be a different cause on my side, it is happening on channel credentials initialization

```
    /data/app/com.tophatch.concepts-1/lib/arm/libgrpc.so:3257717 grpc_combiner_continue_exec_ctx()
    /data/app/com.tophatch.concepts-1/lib/arm/libgrpc.so:3274409 grpc_core::ExecCtx::Flush()
    /data/app/com.tophatch.concepts-1/lib/arm/libgrpc++.so:554541 grpc::SecureChannelCredentials::~SecureChannelCredentials()
    /data/app/com.tophatch.concepts-1/lib/arm/libgrpc++.so:555861 std::__ndk1::__shared_ptr_pointer<grpc::SecureChannelCredentials*, std::__ndk1::default_delete<grpc::SecureChannelCredentials>, std::__ndk1::allocator<grpc::SecureChannelCredentials> >::__on_zero_shared()
```

on this line

```
grpc::SslCredentialsOptions               credentialOptions;
std::shared_ptr<grpc::ChannelCredentials> channel_creds = grpc::SslCredentials(credentialOptions);
```

**Comment by rochaudhari** [comment]
seems like your `channel_creds` is getting destroyed and it is being used somewhere else. Maybe share your code snippet here.

**Comment by cosmin42** [comment]
```
    grpc::SslCredentialsOptions credentialsOptions;
    std::shared_ptr<grpc::ChannelCredentials> channel_creds = grpc::SslCredentials(credentialsOptions);
    if (!channel_creds)
    {
       // return error
    }

    channel  = grpc::CreateCustomChannel(address.c_str(), channel_creds, args);
```
After grpc::CreateCustomChannel, the function which contains this code ends so channel_creds is destroyed and not used anymore.

**Comment by rochaudhari** [comment]
yeah, this part seems okay. did you actually check line number from your source code?

**Comment by cosmin42** [comment]
Yes, I'm thinking that this can be a bug specific for arm architecture as it happens on an Android device with arm architecture and API level 25 which is a bit old.

Thanks!

**Comment by IkshuDM** [comment]
@cosmin42 ,
The crash issue in your system was related to ARM or  was it some race condition. We are facing same issue on issue 1.76 grpc version.
